### PR TITLE
Add support for PRs from forked repos

### PR DESCRIPTION
**Pull request by jtcovan**
This change mainly adds support for PRs from forked repos. 
I did this by adding forked repos as remotes named after their owner (PR author) and fetching their branches to `origin/conflictbot_tmp_`. These tmp branches get cleaned up after each merge attempt so adding them to origin does not cause conflicts.

Some other minor changes:
- Moved the functions related to making api requests to `index.requests.js` to break up `index.js`.
- Added additional logging.
- Change the temporary branch prefix from `tmp` to `conflictbot_tmp`.
- Bug fix: Paginate to retrieve PR files.
- Update to node20 since node12 is deprecated.
- Added periods at the end of comments.

I manually tested this against my own repo. Action runs at: https://github.com/jtcovan/test_bots/actions

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,51 +3,19 @@ const github = require("@actions/github");
...
+  getChangedFiles,
+  requestReviews,
+  requestReviewsInConflictingPRs,
+  createConflictComment,
+} = require("./index.requests");
 
 async function main() {
   if (github.context.payload.pull_request.draft) {
-    debug(`Not running any checks because this pull request is a draft.`)
+    debug(`Not running any checks because this pull request is a draft`)
```
Comment (FGasper) [inline index.js:18]: Why remove the trailing period?

File: index.js [hunk 2]
```diff
@@ -183,7 +111,7 @@ async function getConflictArrayData() {
         title: otherPullRequest.title,
         reviewers: otherPullRequest.reviewers,
       });
-    }
+    }
```
Comment (FGasper) [inline index.js]: trailing space … care to remove?

**Review by FGasper — APPROVED** [review]
LGTM % a couple trivial notes.

My review was pretty surface-level. I’m OK to approve from that since this seems pretty low-risk. If you’d like a more in-depth review, would it be possible to reduce the amount of moved code?

File: index.js (hunk 1 above)
Comment (jtcovan) [reply index.js:18]: This is to match the rest of the debug() strings, which do not end in periods.

**Pull request merged**